### ISSUE TITLE
Loss calculation should indicate which loss(es) returned NaN

### ISSUE THREAD

**Issue by pokepress** [muslll/neosr#87]
I've gotten NaN losses, even at high iteration numbers and low learning rates, so it would be nice to know which one(s) caused it to try continuing with it disabled.

**Comment by muslll** [comment]
Hi. Unfortunately as the current training loop is it's no possible to say exactly which one NaN'd, since all losses are summed before step(). Also, even if it detected NaN in a specific loss, it doesn't necessarily means disabling it will make others stable.
Generally speaking, if you're having NaN there's something wrong on your configuration file or the way you're training. Remember to use a pretrained model and the default template configs at first, to avoid common mistakes.

**Comment by pokepress** [comment]
Well, for the record, I was somewhere between 380k and 385k iterations, and wavelet guiding seems to be the likely culprit since I temporarily turned it off and was able to make it further (390k), then turned it back on and was able to train to 395k.

**Comment by muslll** [comment]
I see, thanks for reporting. And yes, wavelet guided can be a bit hard to use. Some discriminators don't deal well with it and ends up hurting the training process more than it benefits.
